R/plotreg.R

```diff
@@ -1,394 +1,372 @@
+            stop("File extension not recognized.")
         }
-      } else {
-        stop(paste("Custom coefficient names must be provided as a list of", 
-            "character vectors."))
-      }
     }
     
-    # remove any of the coefficients? apply regex
-    remove.rows <- grep(omit.coef, models[[i]]@coef.names)
-    if (is.na(omit.coef)) {
-      remove.rows <- length(models[[i]]@coef) + 1
-    }
+    co <- se <- co.names <- pv <- lab <- ci.low <- ci.up <- NULL
+    # make dataframe
+    for (i in 1:length(models)) {
+        
+        # custom coefficient names
+        if (!is.null(custom.coef.names)) {
+            if (class(custom.coef.names) == "list") {
+                if (length(custom.coef.names[[i]]) == length(models[[i]]@coef.names)) {
+                    models[[i]]@coef.names <- custom.coef.names[[i]]
+                } else {
+                    stop(paste0("Model ", i, 
+                                ": wrong number of custom coefficient names."))
+                }
+            } else if (class(custom.coef.names) == "character") {
+                if (length(custom.coef.names) == length(models[[i]]@coef.names)) {
+                    models[[i]]@coef.names <- custom.coef.names
+                } else {
+                    stop(paste0("Model ", i, 
+                                ": wrong number of custom coefficient names."))
+                }
+            } else {
+                stop(paste("Custom coefficient names must be provided as a list of", 
+                           "character vectors."))
+            }
+        }
+        
+        coef.names <- models[[i]]@coef.names
+        co.names <- append(co.names, coef.names)
+        coef <- models[[i]]@coef
+        co <- append(co, coef)
+        label <- rep(paste0("Model", i),length(models[[i]]@coef))
+        lab <- append(lab, label)
+        if (length(models[[i]]@se) > 0) {
+            s.e <- models[[i]]@se
+            se <- append(se, s.e)
+        } 
+        if(length(models[[i]]@pvalues) > 0) {
+            pvalues <- models[[i]]@pvalues
+            pv <- append(pv, pvalues)
+        }
+        if(length(models[[i]]@ci.low) > 0) {
+            ci.lower <- models[[i]]@ci.low
+            ci.low <- append(ci.low, ci.lower)
+        }
+        if(length(models[[i]]@ci.up) > 0) {
+            ci.upper <- models[[i]]@ci.up
+            ci.up <- append(ci.up, ci.upper)
+        }
+    }    
+    dataframe <- data.frame(cbind(co.names, co, lab ))
+    
+    if (length(models[[i]]@se) > 0) {
+        dataframe <- cbind(dataframe, se)
+    } 
+    
+    if (length(models[[i]]@pvalues) > 0) {
+        dataframe <- cbind(dataframe, pv)
+    } 
+    
+    if (length(models[[i]]@ci.low) > 0) {
+        dataframe <- cbind(dataframe, ci.low)
+    } 
+    
+    if (length(models[[i]]@ci.up) > 0) {
+        dataframe <- cbind(dataframe, ci.up)
+    } 
+    
+    dataframe$co <- as.numeric(as.character(dataframe$co))
```

Comment:
Because co is stored as a factor before that point and if I don't set it as a character is overriding the coefficients with the corresponding integers.

Reply:
OK!